### PR TITLE
BREAKING CHANGE: US135599 Lit 2 migration - Phase 2

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US135599](https://rally1.rallydev.com/#/?detail=/userstory/623256588773&fdp=true): card-overlay, immersive-nav, ou-filter, pagination, toaster Lit 2 migration - Phase 2

This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**.

Breaking change checklist: 

- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used)
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement 
- [x] Remove Dependabot Lit rules in `.github/dependabot.yml`
- [x] Bump any dependencies to their Lit 2 versions
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers (Eg. `this._subscriptionController.hostConnected()`, etc)
- [x] backwards-compatible changes merged

Renamed API's:
- [x] `UpdatingElement` -> `ReactiveElement`
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators)
- [x] `static getStyles()` -> `static finalizeStyles(styles)`
- [x] `_getUpdateComplete()` -> `getUpdateComplete()`
- [x] `NodePart` -> `ChildPart`

Testing checklist: 
- [x] CI/ unit tests pass
- [x] Local testing on demo pages
- [x] Testing component in the LMS

For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,8 +42,8 @@
     "access": "public"
   },
   "dependencies": {
-    "@brightspace-ui/core": "^1",
+    "@brightspace-ui/core": "^2",
     "d2l-navigation": "BrightspaceUI/navigation#semver:^4",
```
Comment (annabelsegalld2l) [inline package.json:46]: https://github.com/BrightspaceUI/navigation is also a lit repo, which needs to be bumped to ^5.

Comment (Vitalii-Misechko) [reply package.json]: Thank you. I'll check these in `package-lock.json`

File: package.json [hunk 2]
```diff
@@ -42,8 +42,8 @@
     "access": "public"
   },
   "dependencies": {
-    "@brightspace-ui/core": "^1",
-    "d2l-navigation": "BrightspaceUI/navigation#semver:^4",
-    "lit-element": "^2"
+    "@brightspace-ui/core": "^2",
```
[the remark this answers is not in the document]
Comment (annabelsegalld2l) [reply package.json:45]: Yep, it has not been released yet. We want all repos to test with what we have now, and prep the PR to include the new versions when they come out. I have a local testing guide that outlines how you can test without the new releases, so failing CI now is expected, until the migration date.

[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply package.json:45]: That's correct. This is @annabelsegalld2l suggestion to simplify migration after merging to `main/master`

File: package.json (hunk 1 above)
Comment (Vitalii-Misechko) [reply package.json]: Fixed

**Review by johngwilkinson — APPROVED** [review]

**Review by anhill-D2L — APPROVED** [review]

**Comment by Vitalii-Misechko** [comment]
Closed in favor of https://github.com/BrightspaceUILabs/immersive-nav/pull/28

**Pull request closed without merging**